app/controllers/admin/schools/participants_controller.rb

```diff
@@ -4,19 +4,18 @@ module Admin
   class Schools::ParticipantsController < Admin::BaseController
     skip_after_action :verify_authorized
     skip_after_action :verify_policy_scoped
-    before_action :set_school
 
     def index
-      @participants = @school.active_ecf_participants.order(:full_name).includes(
-        early_career_teacher_profile: [:mentor, school_cohort: %i[school cohort]],
-        mentor_profile: [school_cohort: %i[school cohort]],
-      )
+      @participant_profiles = policy_scope(ParticipantProfile::Ecf, policy_scope_class: ParticipantProfilePolicy::Scope)
+        .where(school_cohort_id: SchoolCohort.where(school: school).select(:id))
+        .includes(:user)
+        .order("users.full_name")
```

Comment:
Would it be nicer to do something like 

```
school_participant_profiles = school.active_ecf_participant_profiles
                                    .includes(:user)
                                    .order("users.full_name")
@participant_profiles = policy_scope(school_participant_profiles, policy_scope_class: ParticipantProfilePolicy::Scope)
```
?

Reply:
TBH, I think I'll be fighting with `active_ecf_participant_profiles` a bit :) Since this is a subset of `ecf_participants` association, we now need to know that there are 2 associations going through the same foreign key that might both return the same records but which preloading logic is different. Another reason to limit number of associations...